app/controllers/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+require 'date'
+
+class UsersController < ApplicationController
+  include UsersHelper
+
+  def new
+    if session[:current_user]
+      redirect_to events_path
+    else
+      @user = User.new
+    end
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      current_user(@user.id)
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  def show
+    if session[:current_user]
+      @user = User.find(params[:id])
+      current_user(@user.id)
+      @date = date_now
+      @events = @user.attended_events
+      @my_events = @user.events
+    else
+      redirect_to sign_in_path
+    end
+  end
```

Comment:
- [ ] instead of getting the past and upcoming events through the views, use the controller instead ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))